xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-rss/src/main/java/org/xwiki/rendering/internal/macro/rss/RssMacro.java

```diff
@@ -112,22 +121,49 @@ public boolean supportsInlineMode()
         return false;
     }
 
+    private void setFeed(SyndFeed feed)
+    {
+        this.execution.getContext().setProperty(FEED_PROPERTY, feed);
+    }
+
+    private SyndFeed getFeed()
+    {
+        return (SyndFeed) this.execution.getContext().getProperty(FEED_PROPERTY);
+    }
+
+    @Override
+    protected ResourceReference getImageReference(RssMacroParameters parameters, String content,
+        MacroTransformationContext context)
+    {
+        if (parameters.isDecoration() && parameters.isImage()) {
+            return new ResourceReference(getFeed().getImage().getUrl(), ResourceType.URL);
+        } else {
+            return super.getImageReference(parameters, content, context);
+        }
+    }
+
+    @Override
+    protected List<? extends Block> getBlockTitle(RssMacroParameters parameters, String content,
+        MacroTransformationContext context)
+    {
+        return generateBoxTitle("rsschanneltitle", getFeed());
```

Comment:
What do you think about doing that only if block title is not set in `parameters` ? I'm thinking that we now expose this parameter and it does not cost much to do something with it instead of completely ignoring it.

Reply:
so basically if the super method returns null I return the generateBlockTitle, right?

Reply:
I was thinking about directly checking `parameters` actually to be safe but I guess it's OK too yes.